src/components/panel/panel.js

```diff
@@ -806,6 +850,75 @@ MdPanelService.prototype.newPanelAnimation = function() {
 
 
 /**
+ * Creates a panel group and adds it to a tracked list of panel groups.
+ * @param groupName {string} Name of the group to create.
+ * @param config (!Object=) Specific configuration object that may contain the
+ *     following properties:
+ *
+ *   - `maxOpen` - `{number=}`: The maximum number of panels that are allowed
+ *     open within a defined panel group.
+ */
+MdPanelService.prototype.newPanelGroup = function(groupName, config) {
+  if (!this._groups[groupName]) {
+    config = config || {};
+    var group = {
+      panels: [],
+      maxOpen: config.maxOpen > 0 ? config.maxOpen : false
+    };
+    this._groups[groupName] = group;
+  }
+};
+
+
+/**
+ * Set the maximum number of panels in a group that can be opened at a given
+ * time.
+ * @param {string} groupName The name of the group to configure.
+ * @param {number|boolean} maxOpen The maximum number of panels that can be
+ *     opened. If false is passed in, the maximum limit will be removed.
+ */
+MdPanelService.prototype.setGroupMaxOpen = function(groupName, maxOpen) {
+  if (this._groups[groupName]) {
+    this._groups[groupName].maxOpen = maxOpen;
+  }
+};
+
+
+/**
+ * Determines if opening a panel within a group will exceed the limit of allowed
+ * open panels. If so, the first panel opened within the group will be closed.
+ * @param {string} groupName The name of the group to check.
+ * @private
+ */
+MdPanelService.prototype._checkGroupMaxOpen = function(groupName) {
+  if (this._groups[groupName] && this._groups[groupName].maxOpen > 0) {
+    var openCount = 0,
+        firstOpenedPanel = null,
+        firstOpenedPanelIndex = null;
+
+    angular.forEach(this._groups[groupName].panels, function(panel, index) {
+      if (panel.isOpened) {
+        openCount++;
+        if (openCount === 1 && !firstOpenedPanel) {
```

Comment:
@crisbeto Not necessarily. Imagine this scenario:

 - The toolbar group has a `maxOpen` of 3 panels.
 - The user opens `panel1`, then opens `panel2`, and when they open `panel3`, `panel1` closes. The new order inside of the group went from `panel1`, `panel2`, `panel3` to `panel2`, `panel3`, `panel1`.
 - If no ID was provided for any of the panels, when the user clicks again on the icon that opened `panel1`, it is `panel4` that gets opened, not `panel1`. Then, `panel2` is closed. The new order is `panel3`, `panel1`, `panel4`, `panel2`.
 - If the user clicks on the icon that opened `panel2` earlier, it will open `panel5` instead. Then, `panel3` is closed. The new order is `panel1`, `panel4`, `panel2`, `panel5`, `panel3`.
 - If the user then clicks to open another panel, `panel6` is created and added to the group, and the first opened is not `panel1`, which is the first panel in the group, it is `panel4`, the second panel in the group. 

That is why I had to have a special iterator. The whole problem came from having to support a group of panels that have an ID, meaning that it is the same panel opening again and again, and a group of panels that do not have an ID and each click is creating a new panel.

Please let me know if that doesn't make sense.